pkgs/development/python-modules/bash-kernel/default.nix

```diff
@@ -45,8 +45,16 @@ buildPythonPackage rec {
     ${python.pythonOnBuildForHost.interpreter} -m bash_kernel.install --prefix $out
   '';
 
-  # no tests
-  doCheck = false;
+  checkPhase = ''
+    # Create a JUPYTER_PATH with the kernelspec
+    export JUPYTER_PATH=$(mktemp -d)
+    mkdir -p $JUPYTER_PATH/kernels/bash
+    echo '{ "language": "bash", "argv": [ "${python}/bin/python", "-m", "bash_kernel", "-f", "{connection_file}" ] }' > $JUPYTER_PATH/kernels/bash/kernel.json
+
+    # Evaluate a test notebook with papermill
+    cd $(mktemp -d)
+    ${python.withPackages (ps: [ps.papermill])}/bin/papermill --kernel bash ${./test.ipynb} out.ipynb
```

Comment:
so this would fail without bashInteractive ? 
I wonder if it should go through passtru.tests or in checkPhase since papermill becomes a new dependency ? without your fixes, the kernel still works but with a funky prompt ? (sry I read a bit fast)

Reply:
The kernel is totally broken without `bashInteractive`. It crashes right away with an error about being unable to parse the prompt.

I don't know about the current best practices regarding `passthru.tests`. It seems like a good idea to always run a smoke check to prevent regressions like the one we currently have.

If we do the `checkPhase` option, Papermill would only be a build-time dependency right? So users getting it from the Nixpkgs cache wouldn't experience any change.